src/app/core/services/authenticate.service.ts

```diff
@@ -0,0 +1,41 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, Observable } from 'rxjs';
+import { ApiService, AuthToken, UserDTO } from '../../../backend-client';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class AuthenticateService {
+  private currentUserSubject: BehaviorSubject<Observable<UserDTO> | null>;
+  public currentUser: Observable<Observable<UserDTO> | null>;
+
+  constructor(private api: ApiService) {
+    this.currentUserSubject = new BehaviorSubject<Observable<UserDTO> | null>(
```

Comment:
this.currentUserSubject = this.api.authControllerCurrentUser().pipe(shareReplay()) อย่าลืมแก้ type ด้วย